ids/node_id.go

```diff
@@ -21,65 +13,15 @@ const (
 var (
 	EmptyNodeID = NodeID{}
 
-	errShortNodeID = errors.New("insufficient NodeID length")
-
 	_ utils.Sortable[NodeID] = NodeID{}
 )
 
-type NodeID ShortID
-
-func (id NodeID) String() string {
-	return ShortID(id).PrefixedString(NodeIDPrefix)
-}
-
-func (id NodeID) Bytes() []byte {
-	return id[:]
-}
-
-func (id NodeID) MarshalJSON() ([]byte, error) {
-	return []byte(`"` + id.String() + `"`), nil
-}
-
-func (id NodeID) MarshalText() ([]byte, error) {
-	return []byte(id.String()), nil
-}
-
-func (id *NodeID) UnmarshalJSON(b []byte) error {
-	str := string(b)
-	if str == nullStr { // If "null", do nothing
-		return nil
-	} else if len(str) <= 2+len(NodeIDPrefix) {
-		return fmt.Errorf("%w: expected to be > %d", errShortNodeID, 2+len(NodeIDPrefix))
-	}
-
-	lastIndex := len(str) - 1
-	if str[0] != '"' || str[lastIndex] != '"' {
-		return errMissingQuotes
-	}
-
-	var err error
-	*id, err = NodeIDFromString(str[1:lastIndex])
-	return err
-}
-
-func (id *NodeID) UnmarshalText(text []byte) error {
-	return id.UnmarshalJSON(text)
+type NodeID struct {
+	ShortNodeID `serialize:"true"`
```

Comment:
If this diff is "forwards safe" I think we should be able to add an additional field (or somehow else change the serialized format of this struct) without breaking a ton of tests.